### PR TITLE
CIV-14146 Service PR - Claimant LR screen: Permission granted by the court

### PR DESCRIPTION
**Pull request by sjamihmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-14146

### Change description ###
Claimant LR screen: Permission granted by the court

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/enums/settlediscontinue/SettleDiscontinueYesOrNoList.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.civil.enums.settlediscontinue;
```
Comment (ShwetaTandel-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/enums/settlediscontinue/SettleDiscontinueYesOrNoList.java:1]: Why didn't we use existing YesOrNO datatype?

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 2]
```diff
@@ -672,9 +673,10 @@ public boolean hasNoOngoingBusinessProcess() {
     private DynamicList claimantWhoIsDiscontinuing;
     private DynamicList discontinuingAgainstOneDefendant;
     private String selectedClaimantForDiscontinuance;
-    private YesOrNo courtPermissionNeeded;
-    private YesOrNo courtPermissionNeededChecked;
-    private YesOrNo isDiscontinuingAgainstBothDefendants;
+    private SettleDiscontinueYesOrNoList courtPermissionNeeded;
+    //private YesOrNo courtPermissionNeededChecked;
```
Comment (ShwetaTandel-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:677]: Extra comment

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java [hunk 3]
```diff
@@ -8,14 +8,18 @@
...
 import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.PermissionGranted;
 import uk.gov.hmcts.reform.civil.model.common.DynamicList;
 import uk.gov.hmcts.reform.civil.model.common.DynamicListElement;
 import uk.gov.hmcts.reform.civil.model.Party;
 import uk.gov.hmcts.reform.civil.sampledata.CallbackParamsBuilder;
 import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
 
+import java.time.LocalDate;
+
```
Comment (ShwetaTandel-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java:22]: Extra empty line

File: src/main/java/uk/gov/hmcts/reform/civil/enums/settlediscontinue/SettleDiscontinueYesOrNoList.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/enums/settlediscontinue/SettleDiscontinueYesOrNoList.java:1]: We had to create a YesNo json in CCD to display the Yes/No vertical way. So, we have created similar object in Service as YesOrNo type is causing issue in building CaseData and midEvent is not triggered. We used a similar approach as done in existing fields. For eg: ExpertReportsSent

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java (hunk 3 above)
Comment (sjamihmcts) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java:22]: Organized imports

**Review by ShwetaTandel-hmcts — DISMISSED** [review]

**Review by Omaira-Melo-Hmcts — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java [hunk 4]
```diff
@@ -115,4 +119,67 @@ void shouldNotPopulateSelectedClaimant_whenClaimNot2v1() {
...
+            caseData.setIsPermissionGranted(SettleDiscontinueYesOrNoList.YES);
+            caseData.setPermissionGrantedComplex(PermissionGranted.builder()
+                                                     .permissionGrantedJudge("Test")
+                                                     .permissionGrantedDate(LocalDate.of(2070, 12, 12))
+                                                     .build());
+
+            CallbackParams params = callbackParamsOf(caseData, MID, PAGE_ID);
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getErrors().size()).isEqualTo(1);
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java:164]: Should have no errors but error size is 1? Is the ERROR_MESSAGE_UNABLE_TO_DISCONTINUE message?
Also can you please assert the error messages?

Comment (sjamihmcts) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java:164]: Added additional assertions

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Pull request merged**